westcoast/src/main/java/frc/commands/ActuateTwoDoubleSolenoids.java

```diff
@@ -0,0 +1,81 @@
+package frc.commands;
+
+import edu.wpi.first.wpilibj.command.Command;
+import edu.wpi.first.wpilibj.DoubleSolenoid;
+import edu.wpi.first.wpilibj.DoubleSolenoid.Value;
+
+public class ActuateTwoDoubleSolenoids extends Command {
+  private final DoubleSolenoid solenoid;
+  private final DoubleSolenoid solenoid2;
+	private final Value direction;
+  private final Value endDirection;
+  public enum state {
+    bothout, flapperout, pusherout, bothin
+  }
+  private state fire = state.bothout;
+  
+  public ActuateTwoDoubleSolenoids(
+    DoubleSolenoid solenoid,
+    DoubleSolenoid solenoid2,
+    Value direction,
+    Value endDirection
+    ) 
+    {
+      this.direction = direction;
+      this.endDirection = endDirection;
+      this.solenoid = solenoid;
+      this.solenoid2 = solenoid2;
+      
+  }
+
+  @Override
+  protected void initialize() {
+  }
+
+  @Override
+  protected void execute() {
```

Comment:
This function is called multiple times while active so:
```
s0 = solenoid
s1 = solenoid2
f = endDirection
r = direction

t0 => s0=f - s1=f
t1 => s0=r - s1=f
t1 => s0=r - s1=f
```
and so forth. Until you deactivate:
```
tn-1 => s0=r - s1=f
tn   => s0=f - s1=f
```
This is the `pusher out` scenario. But other scenarios follow a similar logic of going to a specific state (`solenoid` go to `direction` and `solenoid2` go to `endDirection`) when activated and go back to the original state when deactivated.

Is that what you are trying to achieve?

Reply:
Ok `both in` will make `s0=r` and `s1=f` and then `s0=f` and `s1=r`.

I confess I'm a little bit confused.
Can you write what is the expected behavior in the PR description?

Reply:
I wrote the expected behavior. As far as I know, the issue was that it would call the other command when interrupted which would make some state functions get angry. However, since I am writing this as a command, I don't know if it will conflict, but if you don't set it up as a "
while held" or have it reset when interrupted or else when the fire button gets called again it must return to the states it wants to be (originally the "both in"). So now it will return to it's uninterrupted position, and stops the command so when it calls it again it goes to the state again. Technically the if/then code isn't needed for the excecute just the end and interrupted, but it still needs to check, the state before and reset it.